MobalMobal/MobalMobal/Signup/SignupViewController.swift

```diff
@@ -0,0 +1,172 @@
+        button.layer.cornerRadius = 30
+        button.backgroundColor = UIColor(red: 71/255, green: 71/255, blue: 71/255, alpha: 1)
+        return button
+    }()
+    
+    private let completeButtonLabel: UILabel = {
+        let label: UILabel = UILabel()
+        label.text = "가입완료"
+        label.font = UIFont(name: "SpoqaHanSansNeo-Medium", size: 18)
+        label.textColor = UIColor(red: 121/255, green: 121/255, blue: 121/255, alpha: 1)
+        return label
+    }()
+    
+    // MARK: - Property
+    private var agreementButtonState: Bool = true
+    
+    // MARK: - Method
+    private func setup() {
+        self.view.backgroundColor = .backgroundColor
+        
+        setUIViewLayout()
+    }
+    
+    @objc
+    func agreementButtonIsTapped(button: UIButton) {
+        agreementButtonState.toggle()
+        
+        if agreementButtonState {
+            button.setImage(UIImage(named: "iconlyLightCheckOff"), for: .normal)
+        } else {
+            button.setImage(UIImage(named: "iconlyLightCheckOn"), for: .normal)
+        }
+    }
+    
+    private func setNicknameView() {
+        self.view.addSubview(nickNameView)
+        self.nickNameView.snp.makeConstraints { make in
+            make.height.equalTo(60)
+            make.top.equalToSuperview().inset(59)
+            make.leading.trailing.equalToSuperview().inset(15)
+        }
+    }
+    
+    private func setPhoneNumberView() {
+        self.view.addSubview(phoneNumberView)
+        self.phoneNumberView.snp.makeConstraints { make in
+            make.height.equalTo(60)
+            make.top.equalTo(nickNameView.snp.bottom).offset(28)
+            make.leading.trailing.equalToSuperview().inset(15)
+        }
+    }
+    
+    private func setEmailView() {
+        self.view.addSubview(emailView)
+        self.emailView.snp.makeConstraints { make in
+            make.height.equalTo(60)
+            make.top.equalTo(phoneNumberView.snp.bottom).offset(28)
+            make.leading.trailing.equalToSuperview().inset(15)
+        }
+    }
+    
+    private func setAgreementView() {
+        self.view.addSubview(agreementView)
+        self.agreementView.snp.makeConstraints { make in
+            make.height.equalTo(44)
+            make.top.equalTo(emailView.snp.bottom).offset(40)
+            make.leading.trailing.equalToSuperview().inset(15)
+        }
+        
+        self.agreementView.addSubview(agreementButton)
+        self.agreementButton.snp.makeConstraints { make in
+            make.centerY.equalTo(self.agreementView)
+            make.leading.equalToSuperview()
+            make.width.equalTo(43)
+        }
+        
+        self.agreementView.addSubview(agreementLabel)
+        self.agreementLabel.snp.makeConstraints { make in
+            make.centerY.equalTo(self.agreementView)
+            make.leading.equalTo(self.agreementButton.snp.trailing).offset(7)
+            make.trailing.equalTo(self.agreementView.snp.trailing).offset(0)
```

Comment:
agreementView가 슈퍼뷰니까 centerY와 trailing을 한 줄에 합칠 수 있습니다!
그리고 제약조건 주체와 대상이 같을 때 (138번줄 trailing) 생략할 수 있어요!

```suggestion
            make.centerY.trailing.equalToSuperview()
            make.leading.equalTo(self.agreementButton.snp.trailing).offset(7)
```

그런데 아래 코드들도 보니까 일부러 순서를 맞추려고 줄을 따로 쓴 것 같기도 하고???
스타일인건가용?!

Reply:
위에랑 비슷한 이유긴 한데 Snapkit 처음쓰면서 조금 더 명시적으로 하다 보니 일단 이렇게 구현했습니다 :)

Reply:
넹넹 그런 이유라면 좋은 것 같슴당!!!😀